### ISSUE TITLE
HeatmapLayer.SetData throws error : b.lat is not a function

### ISSUE THREAD

**Issue by jvakos** [rungwiroon/BlazorGoogleMaps#56]

**Comment by jvakos** [comment]
I found this https://stackoverflow.com/a/58530025/1327555 that assumes LatLngLitteral must not be used in heatmap data. LatLng is the way. I tried for  but I cannot find the point in code that

**Comment by valentasm1** [comment]
Could you post more code? I want to reproduce.

**Comment by jvakos** [comment]
In Maps.razor (server-side project)
` var heatMap = await HeatmapLayer.CreateAsync(JsRuntime, new HeatmapLayerOptions {
                Map = map1,
                Dissipating=true,
                Radius=10,                
            });

            var heatPoints = new[] {
                new LatLngLiteral{                        
                            Lat = 15,
                            Lng = 15,                                                
                },
                new LatLngLiteral{                        
                            Lat = 25,
                            Lng = 25,                                                
                },
            };

            await heatMap.SetData(heatPoints);`

**Comment by valentasm1** [comment]
We serialize .net object to string and pass to object constructor. I have no problem with properties. Since LatLng uses method to return value this way will not work. I added commit to work with SetData with custom coordinates and added if in main method to convert to good format. Get solution and check server side sample. I added sample page with heatmap.
Check commit and let me know if you will make pull request with fixed rest parts? I could do it till September since i take some time.

https://github.com/rungwiroon/BlazorGoogleMaps/commit/d0409c5442379f4a7a78438b0cdb433fbe24238e
